bindings/java/src/test/com/apple/foundationdb/test/RangeScanBenchmark.java

```diff
@@ -0,0 +1,705 @@
+ */
+package com.apple.foundationdb.test;
+
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.apple.foundationdb.Database;
+import com.apple.foundationdb.FDB;
+import com.apple.foundationdb.KeyValue;
+import com.apple.foundationdb.Range;
+import com.apple.foundationdb.StreamingMode;
+import com.apple.foundationdb.Transaction;
+import com.apple.foundationdb.EventKeeper;
+import com.apple.foundationdb.EventKeeper.Events;
+import com.apple.foundationdb.test.ContinuousSample.DoubleSample;
+import com.apple.foundationdb.test.ContinuousSample.LongSample;
+import com.apple.foundationdb.tuple.ByteArrayUtil;
+
+/**
+ * A benchmarking tool for running a variety of different RangeQuery configurations
+ * on a single client machine.
+ *
+ * There are several dimensions to cover in this benchmark, in two different categories:
+ *
+ * 1. Data configuration
+ * 2. Query configuration
+ *
+ * In the category of Data configuration, we want to know:
+ *
+ * 1. number of records in database
+ * 2. total size of each record (or distribution, if you want it to be non-uniform)
+ *
+ * In the category of query configuration, we want:
+ *
+ * 1. number of records in scan
+ * 2. byte limit configuration(i.e. TargetBytesStrategy)
+ * 3. number of concurrent scans
+ */
+public class RangeScanBenchmark {
+
+	public static void main(String... args) throws Exception {
+		FDB fdb = FDB.selectAPIVersion(630);
+		fdb.enableDirectBufferQuery(true);
+
+		MapTimer timer = new MapTimer();
+		RunStatistics statistics = new RunStatistics(timer);
+		StatisticsFormat formatter = new SimpleComparisonFormat();
+		try (Database db = fdb.open(null, timer)) {
+			DataConfiguration config = configureDataLoad();
+			formatter.printDataSetup("Load" + config.numRows, config);
+			if (config != null) {
+				loadData(db, config);
+			}
+
+			Collection<QueryConfiguration> configuration = configureQueryRuns(config);
+
+			int runCnt = 0;
+			for (QueryConfiguration queryConfig : configuration) {
+				runQueryTest("queryTest" + runCnt, db, config, queryConfig, statistics, formatter);
+				runCnt++;
+			}
+
+			if (config.cleanUpAfter()) {
+				cleanUp(db, config);
+			}
+		}
+	}
+
+	private static DataConfiguration configureDataLoad() {
+		return new DataConfiguration(150, 1000, new Random(), true);
+	}
+
+	private static Collection<QueryConfiguration> configureQueryRuns(DataConfiguration dataCfg) {
+		List<QueryConfiguration> configs = new ArrayList<>();
+		for (int rowSize = 0; rowSize <= dataCfg.numRows; rowSize += 5) {
+			configs.add(new QueryConfiguration(1, 50, rowSize, false, StreamingMode.ITERATOR));
+		}
+		return configs;
```

Comment:
These are commented out because they are useful tests that could be made possible by PR #4415.